core/toolbox/toolbox_item.js

```diff
@@ -0,0 +1,226 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview An item in the toolbox.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+'use strict';
+
+goog.provide('Blockly.CollapsibleToolboxItem');
+goog.provide('Blockly.SelectableToolboxItem');
+goog.provide('Blockly.ToolboxItem');
+
+
+/**
+ * Class for an item in the toolbox.
+ * @param {!Blockly.utils.toolbox.ToolboxItemDef} toolboxItemDef The JSON defining the
+ *     toolbox item.
+ * @param {!Blockly.IToolbox} toolbox The toolbox that holds the toolbox item.
+ * @constructor
+ * @abstract
+ */
+Blockly.ToolboxItem = function(toolboxItemDef, toolbox) {
+
+  /**
+   * The id for the category.
+   * @type {string}
+   * @protected
+   */
+  this.id_ = toolboxItemDef['id'] || Blockly.utils.genUid();
+
+  /**
+   * The JSON definition of the toolbox item.
+   * @type {!Blockly.utils.toolbox.ToolboxItemDef}
+   * @protected
+   */
+  this.toolboxItemDef_ = toolboxItemDef;
+
+  /**
+   * The toolbox this category belongs to.
+   * @type {!Blockly.IToolbox}
+   * @protected
+   */
+  this.parentToolbox_ = toolbox;
+
+  /**
+   * The workspace of the parent toolbox.
+   * @type {!Blockly.WorkspaceSvg}
+   * @protected
+   */
+  this.workspace_ = this.parentToolbox_.getWorkspace();
+};
+
+/**
+ * Creates the dom for a toolbox item.
+ * @return {?Element} The div for the toolbox item.
+ * @public
+ */
+Blockly.ToolboxItem.prototype.createDom = function() {
+  return null;
+};
+
+/**
+ * Gets the div for the toolbox item.
+ * @return {?Element} The div for the toolbox item.
+ * @public
+ */
+Blockly.ToolboxItem.prototype.getDiv = function() {
+  return null;
```

Comment:
Since this can be null, does that mean we need to check for nullability here: https://github.com/google/blockly/blob/toolbox/core/toolbox/toolbox.js#L324

Reply:
Yup! Done.